app/objects/leaderboard.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import asyncio
+from typing import Optional
+from typing import TypedDict
+
+import app.state
+from app.constants.gamemodes import GameMode
+from app.objects.beatmap import Beatmap
+from app.objects.score import Score
+
+
+class UserScore(TypedDict):
+    score: Score
+    rank: int
+
+
+class Leaderboard:
+    def __init__(self, mode: GameMode) -> None:
+        self.mode = mode
+        self.scores: list[Score] = []
+
+        self.lock: asyncio.Lock = asyncio.Lock()
+
+    def __len__(self) -> int:
+        return len(self.scores)
+
+    @classmethod
+    async def create_leaderboard(cls, mode: GameMode, beatmap: Beatmap) -> Leaderboard:
+        """Create a leaderboard object with populated scores."""
+
+        leaderboard = Leaderboard(mode)
+
+        rows = await app.state.services.database.fetch_all(
+            "SELECT id, map_md5, userid, pp, score, "
+            "max_combo, mods, acc, n300, n100, n50, "
+            "nmiss, ngeki, nkatu, grade, perfect, "
+            "status, mode, play_time, "
+            "time_elapsed, client_flags, online_checksum "
+            "FROM scores WHERE map_md5 = :map_md5 AND status = 2"
+            "AND mode = :mode",
+            {"map_md5": beatmap.md5, "mode": mode.value},
+        )
+
+        for row in rows:
+            score_obj = Score.from_row(row, calculate_rank=False)
+            leaderboard.scores.append(score_obj)
+
+        await leaderboard.sort()
+        return leaderboard
+
+    async def remove_score_index(self, index: int) -> None:
+        async with self.lock:
+            self.scores.pop(index)
+
+    async def find_user_score(
+        self,
+        user_id: int,
+    ) -> Optional[UserScore]:
```

Comment:
```suggestion
    async def find_user_score(self, user_id: int) -> Optional[UserScore]:
```

Reply:
black 👍🏻

Reply:
if u add the `,`, it will do that - this is probably the only control you have with black